relayer/application.ts

```diff
@@ -95,6 +110,42 @@ export class RelayerApp<ContextT extends Context> extends EventEmitter {
     this.opts = mergeDeep({}, [defaultOpts(env), opts]);
   }
 
+  /**
+   *  This function will run as soon as a VAA is received and will determine whether we want to process it or skip it.
+   *  This is useful if you're listening to a contract but you don't care about every one of the VAAs emitted by it (eg. The Token Bridge contract).
+   *
+   *  WARNING: If your function throws, the VAA will be skipped (is this the right behavior?). If you want to process the VAA anyway, catch your errors and return true.
+   *
+   * @param newFilter pass in a function that will receive the raw bytes of the VAA and if it returns `true` or `Promise<true>` the VAA will be processed, otherwise it will be skipped
+   */
+  filter(newFilter: FilterFN) {
+    const previousFilters = this.shouldProcessVaa;
+    this.shouldProcessVaa = async (vaaBytes: ParsedVaaWithBytes) => {
+      try {
+        let [should1, should2] = await Promise.all([
```

Comment:
Logging to determine which filter caused the vaa to be dropped?

Reply:
I was hoping the filter method you pass in would do the logging explaining why it returns false, otherwise what do I log? I can log the fn name but it may be an anonymous function and you wouldn't get much out of that.

Reply:
I was thinking something like:
> logger.error(`Filter function ${i} of ${n} errored`, {error: e}) 
Or catch the exception and add the filter index:
```typescript
} catch (e) {
     e.filterIdx = i
     e.numFilters = n
     throw e
}
```